common/buildcraft/transport/stripes/PipeExtensionManager.java

```diff
@@ -0,0 +1,176 @@
+                    }
+                }
+
+                // Step 1: Copy over and remove existing stripes pipe
+                IBlockState stripesStateOld = w.getBlockState(r.pos);
+                NBTTagCompound stripesNBTOld = new NBTTagCompound();
+                TileEntity stripesTileOld = w.getTileEntity(r.pos);
+                if (!stripesTileOld.hasCapability(PipeApi.CAP_PIPE_HOLDER, null)) {
+                    BCLog.logger.warn("Found an invalid request at " + r.pos + " as " + stripesTileOld + " was not a pipe tile!");
+                    continue;
+                }
+                stripesTileOld.writeToNBT(stripesNBTOld);
+                w.setBlockToAir(r.pos);
+
+                // Step 2: If retracting, remove previous pipe; if extending, add new pipe
+                if (retract) {
+                    IBlockState state = w.getBlockState(p);
+                    List<ItemStack> list = state.getBlock().getDrops(w, p, state, 0);
+                    w.setBlockToAir(p);
+                    stacksToSendBack.add(r.stack);
+                    if (list != null) {
+                        stacksToSendBack.addAll(list);
+                    }
+                } else {
+                    FakePlayer player = BuildCraftAPI.fakePlayerProvider.getFakePlayer((WorldServer) w, null, r.pos);
+                    player.inventory.clear();
+                    player.inventory.setInventorySlotContents(player.inventory.currentItem, r.stack);
+                    BlockUtil.useItemOnBlock(w, player, r.stack, r.pos, r.dir.getOpposite());
+                    for (int i = 0; i < player.inventory.getSizeInventory(); i++) {
+                        ItemStack stack = player.inventory.removeStackFromSlot(i);
+                        if (!stack.isEmpty()) {
+                            stacksToSendBack.add(stack);
+                        }
+                    }
+                }
+
+                // Step 3: Place stripes pipe back
+                // - Correct NBT coordinates
+                stripesNBTOld.setInteger("x", p.getX());
+                stripesNBTOld.setInteger("y", p.getY());
+                stripesNBTOld.setInteger("z", p.getZ());
+                // - Create block and tile
+                w.setBlockState(p, stripesStateOld, 3);
+
+                TileEntity stripesTileNew = w.getTileEntity(p);
+                stripesTileNew.setWorld(w);
+                stripesTileNew.readFromNBT(stripesNBTOld);
+                stripesTileNew.onLoad();
+
+                if (stripesTileNew.hasCapability(PipeApi.CAP_PIPE_HOLDER, null)) {
+                    IPipeHolder stripesPipeHolderNew = stripesTileNew.getCapability(PipeApi.CAP_PIPE_HOLDER, null);
+                    IWireManager wireManager = stripesPipeHolderNew.getWireManager();
+                    if (wireManager instanceof WireManager) {
+                        ((WireManager) wireManager).getWireSystems().rebuildWireSystemsAround(stripesPipeHolderNew);
```

Comment:
Any way in the API to do this?

Reply:
This is kinda ok, we can assume that every pipe tile is an TilePipeHolder which uses WireManager